### PR TITLE
SDIT-1358 Remove SYSTEM_USER from remaining Resource files:

### PR DESCRIPTION
**Pull request by steverendell**
GET /api/offenders/{offenderNo}/case-notes/v2
GET /api/offenders/{offenderNo}/case-notes/{caseNoteId}
GET /api/offenders/{offenderNo}/contacts
GET /api/offenders/{offenderNo}/offender-restrictions

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 1]
```diff
@@ -478,7 +478,11 @@ public PrisonerIdentifier getNextPrisonerIdentifier() {
...
+    public List<IncidentCase> getIncidentsByOffenderNo(
+        @PathVariable("offenderNo") @Parameter(description = "offenderNo", required = true, example = "A1234AA") @NotNull final String offenderNo,
+        @RequestParam("incidentType") @Parameter(description = "incidentType", example = "ASSAULT") final List<String> incidentTypes,
+        @RequestParam("participationRoles") @Parameter(description = "participationRoles", example = "ASSIAL", schema = @Schema(implementation = String.class, allowableValues = {"ACTINV", "ASSIAL", "FIGHT", "IMPED", "PERP", "SUSASS", "SUSINV", "VICT", "AI", "PAS", "AO"})) final List<String> participationRoles
+    ) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:485]: Just formatting

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 2]
```diff
@@ -505,14 +509,16 @@ public List<AddressDto> getAddressesByOffenderNo(@PathVariable("offenderNo") @Pa
...
+        @RequestParam(value = "finding", required = false) @Parameter(description = "Finding code to allow optionally filtering by type of finding", example = "NOT_PROVEN") final String finding,
+        @RequestParam(value = "fromDate", required = false) @DateTimeFormat(iso = DATE) @Parameter(description = "Adjudications must have been reported on or after this date (in YYYY-MM-DD format).") final LocalDate fromDate,
+        @RequestParam(value = "toDate", required = false) @DateTimeFormat(iso = DATE) @Parameter(description = "Adjudications must have been reported on or before this date (in YYYY-MM-DD format).") LocalDate toDate,
+        @RequestHeader(value = "Page-Offset", defaultValue = "0", required = false) @Parameter(description = "Requested offset of first record in returned collection of adjudications.") final Long pageOffset,
+        @RequestHeader(value = "Page-Limit", defaultValue = "10", required = false) @Parameter(description = "Requested limit to number of adjudications returned.") final Long pageLimit
+    ) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:521]: Just formatting

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 3]
```diff
@@ -657,7 +665,11 @@ public CaseNote createOffenderCaseNote(@PathVariable("offenderNo") @Parameter(de
     @PutMapping("/{offenderNo}/case-notes/{caseNoteId}")
     @HasWriteScope
     @ProxyUser
-    public CaseNote updateOffenderCaseNote(@PathVariable("offenderNo") @Parameter(description = "Noms ID or Prisoner number (also called offenderNo)", required = true, example = "A1234AA") final String offenderNo, @PathVariable("caseNoteId") @Parameter(description = "The case note id", required = true, example = "1212134") final Long caseNoteId, @RequestBody @Parameter(required = true) final UpdateCaseNote body) {
+    public CaseNote updateOffenderCaseNote(
+        @PathVariable("offenderNo") @Parameter(description = "Noms ID or Prisoner number (also called offenderNo)", required = true, example = "A1234AA") final String offenderNo,
+        @PathVariable("caseNoteId") @Parameter(description = "The case note id", required = true, example = "1212134") final Long caseNoteId,
+        @RequestBody @Parameter(required = true) final UpdateCaseNote body
+    ) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:672]: Just formatting

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 4]
```diff
@@ -692,7 +704,10 @@ public SentenceSummary getLatestSentenceSummary(@PathVariable("offenderNo") @Par
...
     @GetMapping("/{offenderNo}/damage-obligations")
-    public OffenderDamageObligationResponse getOffenderDamageObligations(@NotNull @PathVariable("offenderNo") @Parameter(description = "offenderNo", required = true, example = "A1234AA") final String offenderNo, @RequestParam(value = "status", required = false, defaultValue = "ALL") @Parameter(description = "Filter by obligation status. Leave blank to return all", example = "ACTIVE", schema = @Schema(implementation = String.class, allowableValues = {"INACT","PAID","ONH","ACTIVE","APPEAL"})) final String status) {
+    public OffenderDamageObligationResponse getOffenderDamageObligations(
+        @NotNull @PathVariable("offenderNo") @Parameter(description = "offenderNo", required = true, example = "A1234AA") final String offenderNo,
+        @RequestParam(value = "status", required = false, defaultValue = "ALL") @Parameter(description = "Filter by obligation status. Leave blank to return all", example = "ACTIVE", schema = @Schema(implementation = String.class, allowableValues = {"INACT", "PAID", "ONH", "ACTIVE", "APPEAL"})) final String status
+    ) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:710]: Just formatting

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 5]
```diff
@@ -754,7 +769,7 @@ public MilitaryRecords getMilitaryRecords(
             @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Offender Contacts", description = "Active Contacts including restrictions, using latest offender booking  and including inactive contacts by default")
-    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER", "OFFENDER_CONTACTS"})
+    @VerifyOffenderAccess(overrideRoles = {"OFFENDER_CONTACTS", "VIEW_CONTACTS"})
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:772]: Some clients do not have OFFENDER_CONTACTS, but do have VIEW_CONTACTS so it seems logical to add this

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**